ultradns/src/main/java/denominator/ultradns/UltraDNSTarget.java

```diff
@@ -58,9 +59,25 @@ public String url() {
 
   @Override
   public Request apply(RequestTemplate in) {
+    String username;
+    String password;
+    Credentials creds = credentials.get();
+    if (creds instanceof  List) {
+      @SuppressWarnings("unchecked")
+      List<Object> listCreds = (List<Object>) creds;
+      username = listCreds.get(0).toString();
+      password = listCreds.get(1).toString();
+    } else if (creds instanceof Map) {
+      @SuppressWarnings("unchecked")
+      Map<String, Object> mapCreds = (Map<String, Object>) creds;
+      username = checkNotNull(mapCreds.get("username"), "username").toString();
+      password = checkNotNull(mapCreds.get("password"), "password").toString();
+    } else {
+      throw new IllegalArgumentException("Unsupported credential type: "+ creds);
+    }
+
     in.insert(0, url());
-    List<Object> creds = ListCredentials.asList(credentials.get());
-    in.body(format(SOAP_TEMPLATE, creds.get(0).toString(), creds.get(1).toString(),
+    in.body(format(SOAP_TEMPLATE, username, password,
```

Comment:
supernit: you can now bring `new String(in.body(), UTF_8)));` onto this line as it no longer needs to wrap